metricbeat/module/windows/services/services_windows.go

```diff
@@ -0,0 +1,374 @@
+		state = stat
+	} else {
+		state = "Can not define State"
+	}
+	service.CurrentState = state
+
+	// Get detailed information
+	if err := getDetailedServiceInfo(handle, service.ServiceName, ServiceQueryConfig, &service); err != nil {
+		return service, err
+	}
+
+	//Get uptime for service
+	if ServiceState(rawService.ServiceStatusProcess.DwCurrentState) != ServiceStopped {
+		processUpTime, err := getServiceUptime(rawService.ServiceStatusProcess.DwProcessId)
+		if err != nil {
+			logp.Warn("Uptime for service %v is not available", service.ServiceName)
+		}
+		service.Uptime = processUpTime
+	}
+
+	return service, nil
+}
+
+// getServiceUptime returns the uptime for process
+func getServiceUptime(processID uint32) (time.Duration, error) {
+	var processCreationTime gosigar.ProcTime
+
+	err := processCreationTime.Get(int(processID))
+	if err != nil {
+		return time.Duration(processCreationTime.StartTime), err
+	}
+
+	tm := time.Now().UnixNano()
+	uptime := time.Duration((tm / int64(time.Millisecond)) - int64(processCreationTime.StartTime))
+
+	return uptime, nil
+}
+
+func getDetailedServiceInfo(handle ServiceDatabaseHandle, serviceName string, accessRight ServiceAccessRight, service *ServiceStatus) error {
+	var serviceBufSize uint32
+	var serviceBytesNeeded uint32
+
+	serviceHandle, err := OpenService(handle, service.ServiceName, ServiceQueryConfig)
+	if err != nil {
+		return err
+	}
+
+	if err := _QueryServiceConfig(serviceHandle, nil, serviceBufSize, &serviceBytesNeeded); err != nil {
+		if ServiceErrno(err.(syscall.Errno)) != SERVICE_ERROR_INSUFFICIENT_BUFFER {
+			if err := CloseServiceHandle(serviceHandle); err != nil {
+				return err
+			}
+			return err
+		}
+		serviceBufSize += serviceBytesNeeded
+		buffer := make([]byte, serviceBufSize)
+
+		for {
+			if err := _QueryServiceConfig(serviceHandle, &buffer[0], serviceBufSize, &serviceBytesNeeded); err != nil {
+				if ServiceErrno(err.(syscall.Errno)) != SERVICE_ERROR_INSUFFICIENT_BUFFER {
+					if err := CloseServiceHandle(serviceHandle); err != nil {
+						return err
+					}
+					return err
+				}
+				serviceBufSize += serviceBytesNeeded
+			} else {
+				serviceQueryConfig := (*QueryServiceConfig)(unsafe.Pointer(&buffer[0]))
+				service.StartType = serviceStartTypes[ServiceStartType(serviceQueryConfig.DwStartType)]
+				if err := CloseServiceHandle(serviceHandle); err != nil {
+					return err
+				}
+				break
+			}
+		}
+	}
+	return nil
+}
+
+func (reader *ServiceReader) Close() error {
+	return CloseServiceDatabaseHandle(reader.handle)
+}
+
+func CloseServiceDatabaseHandle(handle ServiceDatabaseHandle) error {
+	if err := _CloseServiceHandle(uintptr(handle)); err != nil {
+		return ServiceErrno(err.(syscall.Errno))
+	}
+
+	return nil
+}
+
+func CloseServiceHandle(handle ServiceHandle) error {
+	if err := _CloseServiceHandle(uintptr(handle)); err != nil {
+		return ServiceErrno(err.(syscall.Errno))
+	}
+
+	return nil
+}
+
+func NewServiceReader() (*ServiceReader, error) {
+
```

Comment:
Remove newline from beginning of method. (BTW I built a tool to find these called [nonewlines](https://github.com/andrewkroh/go-nonewlines).)